tests/e2e/dev.rs

```diff
@@ -0,0 +1,98 @@
+use std::env;
+use std::process::Command;
+use std::time::Duration;
+
+use assert_cmd::prelude::CommandCargoExt;
+use mime::APPLICATION_JSON;
+use reqwest::header::CONTENT_TYPE;
+use reqwest::Client;
+use rstest::*;
+use serde_json::{json, Value};
+use speculoos::assert_that;
+use tempfile::TempDir;
+use tokio::time::timeout;
+
+use crate::e2e::{
+    run_subgraphs_retail_supergraph, test_graphql_connection, GRAPHQL_TIMEOUT_DURATION,
+};
+
+const ROVER_DEV_TIMEOUT: Duration = Duration::from_secs(45);
+const ROUTER_PORT: u32 = 4123;
+
+#[fixture]
+#[once]
+fn run_rover_dev(#[from(run_subgraphs_retail_supergraph)] working_dir: &TempDir) -> String {
```

Comment:
Should we have a bespoke return type that implements `Drop` here?

Reply:
Discussed in Slack here: https://apollograph.slack.com/archives/C06HGDC3W3Z/p1721901865421839